framework/hpp_api_vulkan_sample.cpp

```diff
@@ -485,6 +485,15 @@ void HPPApiVulkanSample::submit_frame()
 			present_info.setWaitSemaphores(semaphores.render_complete);
 		}
 
+		vk::DisplayPresentInfoKHR disp_present_info{};
+		if (device->is_extension_supported(VK_KHR_DISPLAY_SWAPCHAIN_EXTENSION_NAME) &&
+		    platform->get_window().get_display_present_info(reinterpret_cast<VkDisplayPresentInfoKHR *>(&disp_present_info),
+		                                                    extent.width, extent.height))
+		{
```

Comment:
Would be great, if you could add some wrapper function to platform/hpp_window.h, like this:
```
	bool get_display_present_info(vk::DisplayPresentInfoKHR* info, uint32_t src_width, uint32_t src_height) const
	{
		return get_display_present_info(reinterpret_cast<VkDisplayPresentInfoKHR *>(info), src_width, src_height);
	}
```
to keep this file free from reinterpreting to VK-structs.

Reply:
Done